qdb/tracer.py

```diff
@@ -145,16 +152,17 @@ def _init(self, config=None, merge=False, **kwargs):
             self.log_handler.push_application()
 
         self.bound_cmd_manager = config.cmd_manager or TerminalCommandManager()
+        self.redirect_output = (
```

Comment:
why did we move this away from the other output stuff?

Reply:
This was again accounting for missing attributes when exceptions were raised in `_init`. I'll look into undoing that, and leaving the singleton change.